src/main/scala/wdl4s/values/WdlOptionalValue.scala

```diff
@@ -0,0 +1,18 @@
+package wdl4s.values
+
+import wdl4s.types.{WdlOptionalType, WdlType}
+
+case class WdlOptionalValue(innerType: WdlType, value: Option[WdlValue]) extends WdlValue {
+  override val wdlType = WdlOptionalType(innerType)
+  override val toWdlString = value map { _.toWdlString } getOrElse "null"
```

Comment:
To answer your first question, there is no possible round trip. You can go `value => value?` but you cannot go `value? => value` without the select_x functions.

The second question, it would give you `hello  world` since the string interpolation is a separate function from `toWdlString` (just like `"echo ${array}"` doesn't print out the WDL array literal like `[1,2,3]`). `null` might be the WDL literal that evaluates to None although that's not-yet-implemented.